packages/data/src/services/open-telemetry/adapter.ts

```diff
@@ -0,0 +1,443 @@
+import type { InputOutputData } from "@evilmartians/agent-prism-types";
+
+import {
+  INPUT_OUTPUT_ATTRIBUTES,
+  OPENINFERENCE_ATTRIBUTES,
+  OPENINFERENCE_MAPPINGS,
+  OPENTELEMETRY_GENAI_ATTRIBUTES,
+  OPENTELEMETRY_GENAI_MAPPINGS,
+  STANDARD_OPENTELEMETRY_ATTRIBUTES,
+  STANDARD_OPENTELEMETRY_PATTERNS,
+  type OpenTelemetryDocument,
+  type OpenTelemetrySpan,
+  type OpenTelemetryStandard,
+  type TraceSpan,
+  type TraceSpanCategory,
+  type TraceSpanStatus,
+} from "@evilmartians/agent-prism-types";
+
+import { SpanAdapter } from "../common/adapter";
+
+export class OpenTelemetrySpanAdapter extends SpanAdapter<
```

Comment:
The naming and semantics here are a bit confusing right now. `SpanAdapter` doesn’t clearly convey its purpose, and the generics aren’t self-explanatory. What do they represent? Should they be input/output types? If so, shouldn’t the output be `UiSpan` instead of `OpenTelemetrySpan`? It would be great if the class and its generics were more descriptive so the intent is clear at a glance.